File: app/src/main/java/com/ternaryop/photoshelf/fragment/FragmentActivityStatus.java
package com.ternaryop.photoshelf.fragment;

import android.support.v7.widget.Toolbar;

import com.ternaryop.photoshelf.AppSupport;


/**
 * Communicate activity status to fragment
 * @author dave
 *
 */
public interface FragmentActivityStatus {
    boolean isDrawerOpen();
    AppSupport getAppSupport();
    Toolbar getToolbar();
}

File: app/src/main/java/com/ternaryop/photoshelf/fragment/FragmentActivityStatus.kt
package com.ternaryop.photoshelf.fragment

import androidx.appcompat.widget.Toolbar

import com.ternaryop.photoshelf.AppSupport

/**
 * Communicate activity status to fragment
 * @author dave
 */
interface FragmentActivityStatus {
    val isDrawerMenuOpen: Boolean
    val appSupport: AppSupport
    val drawerToolbar: Toolbar
}
